protocol/messenger_peersyncing.go

```diff
@@ -164,12 +182,48 @@ func (m *Messenger) sendDatasyncOffers() error {
 		if err != nil {
 			return err
 		}
+	}
+	return nil
+}
+
+func (m *Messenger) sendDatasyncOffersForChats() error {
+	for _, chat := range m.Chats() {
+		availableMessages, err := m.peersyncing.AvailableMessagesByChatID([]byte(chat.ID), maxAdvertiseMessages)
+		if err != nil {
+			return err
+		}
+		availableMessagesMap := make(map[string][][]byte)
+		for _, m := range availableMessages {
+			chatID := types.Bytes2Hex(m.ChatID)
```

Comment:
Is `chatID` the same in all iterations?

Reply:
No it can be different. I've updated it to avoid the confusion.